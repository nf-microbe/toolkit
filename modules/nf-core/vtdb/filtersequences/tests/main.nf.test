nextflow_process {

    name "Test Process: VTDB_FILTERSEQUENCES"
    script "../main.nf"
    process "VTDB_FILTERSEQUENCES"

    tag "modules_nfcore"
    tag "modules"
    tag "vtdb"
    tag "vtdb/filtersequences"

    test("fasta.gz + faa.gz + combined_data.tsv + sequences_to_keep.tsv") {
        when {
            process {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test' ],
                        file("${projectDir}/assets/test-datasets/modules/filtersequences/test.fasta.gz", checkIfExists: true)
                    ]
                )
                input[1] = Channel.of(
                    [
                        [ id:'test' ],
                        file("${projectDir}/assets/test-datasets/modules/filtersequences/test.faa.gz", checkIfExists: true)
                    ]
                )
                input[2] = Channel.of(
                    [
                        [ id:'test' ],
                        file("${projectDir}/assets/test-datasets/modules/filtersequences/test_completeness_data.tsv", checkIfExists: true)
                    ]
                )
                input[3] = Channel.of(
                    [
                        file("${projectDir}/assets/test-datasets/modules/filtersequences/test_sequences_to_keep.tsv", checkIfExists: true)
                    ]
                )
                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("fasta.gz + faa.gz + combined_data.tsv + no sequences to keep") {
        when {
            process {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test' ],
                        file("${projectDir}/assets/test-datasets/modules/filtersequences/test.fasta.gz", checkIfExists: true)
                    ]
                )
                input[1] = Channel.of(
                    [
                        [ id:'test' ],
                        file("${projectDir}/assets/test-datasets/modules/filtersequences/test.faa.gz", checkIfExists: true)
                    ]
                )
                input[2] = Channel.of(
                    [
                        [ id:'test' ],
                        file("${projectDir}/assets/test-datasets/modules/filtersequences/test_completeness_data.tsv", checkIfExists: true)
                    ]
                )
                input[3] = Channel.of([])
                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("stub") {
        when {
            process {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test' ],
                        file("${projectDir}/assets/test-datasets/modules/filtersequences/test.fasta.gz", checkIfExists: true)
                    ]
                )
                input[1] = Channel.of(
                    [
                        [ id:'test' ],
                        file("${projectDir}/assets/test-datasets/modules/filtersequences/test.faa.gz", checkIfExists: true)
                    ]
                )
                input[2] = Channel.of(
                    [
                        [ id:'test' ],
                        file("${projectDir}/assets/test-datasets/modules/filtersequences/test_completeness_data.tsv", checkIfExists: true)
                    ]
                )
                input[3] = Channel.of(
                    [
                        file("${projectDir}/assets/test-datasets/modules/filtersequences/test_sequences_to_keep.tsv", checkIfExists: true)
                    ]
                )
                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
