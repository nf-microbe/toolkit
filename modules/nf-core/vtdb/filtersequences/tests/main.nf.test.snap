{
    "fasta.gz + faa.gz + combined_data.tsv + no sequences to keep": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered.fasta.gz:md5,8bb9e936bc24116c8b538ba95af9c482"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered_proteins.faa.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,309f91bb81b90968be3a91527ebd115c"
                ],
                "fasta": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered.fasta.gz:md5,8bb9e936bc24116c8b538ba95af9c482"
                    ]
                ],
                "proteins": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered_proteins.faa.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,309f91bb81b90968be3a91527ebd115c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-08T08:47:30.877305875"
    },
    "stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered.fasta.gz:md5,8bb9e936bc24116c8b538ba95af9c482"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered_proteins.faa.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,309f91bb81b90968be3a91527ebd115c"
                ],
                "fasta": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered.fasta.gz:md5,8bb9e936bc24116c8b538ba95af9c482"
                    ]
                ],
                "proteins": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered_proteins.faa.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,309f91bb81b90968be3a91527ebd115c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-08T08:47:41.137426022"
    },
    "fasta.gz + faa.gz + combined_data.tsv + sequences_to_keep.tsv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered.fasta.gz:md5,8bb9e936bc24116c8b538ba95af9c482"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered_proteins.faa.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,309f91bb81b90968be3a91527ebd115c"
                ],
                "fasta": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered.fasta.gz:md5,8bb9e936bc24116c8b538ba95af9c482"
                    ]
                ],
                "proteins": [
                    [
                        {
                            "id": "test"
                        },
                        "test_filtered_proteins.faa.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,309f91bb81b90968be3a91527ebd115c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-08T08:47:20.439600016"
    }
}